### PR TITLE
use modern python versions

### PR DESCRIPTION
**Pull request by Jimexist**

### CODE REVIEW

File: .github/workflows/integration-parquet.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
           python -m venv venv
           source venv/bin/activate
           pip install --upgrade pip
-          pip install pyarrow pyspark
+          pip install pyarrow==6.0.1 pyspark==3.2.0
```
Comment (jorgecarleitao) [inline .github/workflows/integration-parquet.yml:49]: I left it open on purpose - to ensure we remain compatible.

Maybe then using 

```suggestion
          pip install pyarrow==6 pyspark==3
```

**Review by jorgecarleitao — APPROVED** [review]
Thanks, @Jimexist !

File: arrow-pyarrow-integration-testing/README.md [hunk 2]
```diff
@@ -35,17 +36,19 @@ we can use pyarrow's interface to move pointers from and to Rust.
...
+- [PyO3](https://docs.rs/pyo3/0.12.1/pyo3/index.html)
 
 ## How to develop
 
 ```bash
 # prepare development environment (used to build wheel / install in development)
 python -m venv venv
-venv/bin/pip install maturin==0.8.2 toml==0.10.1 pyarrow==1.0.0
+source venv/bin/activate
+pip install maturin==0.12.6 toml==0.10.2 pyarrow==6.0.1
```
Comment (Jimexist) [inline arrow-pyarrow-integration-testing/README.md:50]:
```suggestion
pip install maturin==0.12.6 toml==0.10.2 pyarrow==6
```

File: .github/workflows/integration-ffi.yml [hunk 3]
```diff
@@ -38,6 +38,6 @@ jobs:
           python -m venv venv
           source venv/bin/activate
 
-          pip install maturin==0.10.2 toml==0.10.1 pyarrow==6.0.0
+          pip install maturin==0.12.6 toml==0.10.2 pyarrow==6.0.1
```
Comment (Jimexist) [inline .github/workflows/integration-ffi.yml:41]:
```suggestion
          pip install maturin==0.12.6 toml==0.10.2 pyarrow==6
```

**Pull request merged**